urdf_parser_plugin/include/urdf_parser_plugin/parser.h

```diff
@@ -54,7 +56,13 @@ class URDFParser
   }
 
   /// \brief Load Model from string
-  virtual urdf::ModelInterfaceSharedPtr parse(const std::string &xml_string) = 0;
+  /// \return nullptr and write to stderr if the given string is invalid
+  virtual urdf::ModelInterfaceSharedPtr parse(const std::string & data) = 0;
+
+  /// \brief Indicate if data is meant to be parsed by this parser
+  /// \return confidence that the data is meant for this parser, where smaller
+  ///         values mean more confident, and 0 is absolute certainty.
```

Comment:
One more small thing to add here: `std::size_t::max` should be used for no confidence.

Reply:
Document size_t max is no confidence in c9f629d60d378407a642d90c196f6ed6958d4220